src/utils/aadUser.ts

```diff
@@ -25,6 +25,47 @@ export const aadUser = {
     return res.value[0].id;
   },
 
+  /**
+   * Retrieve the IDs of users by their UPNs. There is no guarantee that the order of the returned IDs will match the order of the specified UPNs.
+   * @param upns Array of user UPNs.
+   * @returns Array of user IDs.
+   */
+  async getUserIdsByUpns(upns: string[]): Promise<string[]> {
+    const userIds: string[] = [];
+
+    for (let i = 0; i < upns.length; i += 20) {
+      const upnsChunk = upns.slice(i, i + 20);
+      const requestOptions: CliRequestOptions = {
+        url: `${graphResource}/v1.0/$batch`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          requests: upnsChunk.map((upn, index) => ({
+            id: index + 1,
+            method: 'GET',
+            url: `/users/${formatting.encodeQueryParameter(upn)}?$select=id`,
+            headers: {
+              accept: 'application/json;odata.metadata=none'
+            }
+          }))
+        }
+      };
+      const res = await request.post<{ responses: { id: number; status: number; body: any }[] }>(requestOptions);
```

Comment:
Shouldn't the body be set to the Graph User type?

Reply:
Hmm, I find this tricky because we use `$select=id`. Maybe it's safer to type it as `{ id: string }`?

Reply:
Aren't all properties marked as optional on Graph types?